File: packages/ts/src/models/sync_transaction.ts
/**
 * SyncTransaction — v2 sync response shape (Plaid-compatible flattened model).
 * Returned in `added` / `modified` arrays by POST /v2/accounts/{id}/transactions/sync.
 */

export interface SyncTransaction {
  transaction_id: string;
  account_id: string;
  amount: number;
  iso_currency_code: string;
  unofficial_currency_code: string;
  date: string;
  datetime: string | null;
  name: string;
  merchant_name: string | null;
  pending: boolean;
  category: string[];
  /** Conditional: only when running balances are enabled. */
  running_balance_cents?: number;
}

export interface SyncRemoved {
  transaction_id: string;
  account_id: string;
}

export interface SyncResult {
  added: SyncTransaction[];
  modified: SyncTransaction[];
  removed: SyncRemoved[];
  next_cursor: string | null;
  has_more: boolean;
}

File: packages/csharp/src/Models/SyncTransaction.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Tesote.Sdk.Models;

/// <summary>Flattened Plaid-shaped transaction returned from POST /v2/.../transactions/sync.</summary>
public sealed record SyncTransaction(
    [property: JsonPropertyName("transaction_id")] string TransactionId,
    [property: JsonPropertyName("account_id")] string AccountId,
    [property: JsonPropertyName("amount")] decimal Amount,
    [property: JsonPropertyName("iso_currency_code")] string IsoCurrencyCode,
    [property: JsonPropertyName("unofficial_currency_code")] string? UnofficialCurrencyCode,
    [property: JsonPropertyName("date")] string Date,
    [property: JsonPropertyName("datetime")] string? Datetime,
    [property: JsonPropertyName("name")] string Name,
    [property: JsonPropertyName("merchant_name")] string? MerchantName,
    [property: JsonPropertyName("pending")] bool Pending,
    [property: JsonPropertyName("category")] IReadOnlyList<string> Category,
    [property: JsonPropertyName("running_balance_cents")] long? RunningBalanceCents);

/// <summary>Identifiers for a transaction removed since the last sync.</summary>
public sealed record SyncRemoved(
    [property: JsonPropertyName("transaction_id")] string TransactionId,
    [property: JsonPropertyName("account_id")] string AccountId);

/// <summary>Response envelope for POST /v2/.../transactions/sync.</summary>
public sealed record SyncResult(
    [property: JsonPropertyName("added")] IReadOnlyList<SyncTransaction> Added,
    [property: JsonPropertyName("modified")] IReadOnlyList<SyncTransaction> Modified,
    [property: JsonPropertyName("removed")] IReadOnlyList<SyncRemoved> Removed,
    [property: JsonPropertyName("next_cursor")] string? NextCursor,
    [property: JsonPropertyName("has_more")] bool HasMore);
